### ISSUE TITLE
The provided view controller is not being presented

### ISSUE THREAD

**Issue by abear247** [googleads/googleads-mobile-ios-examples#374]
This is likely a Swift bug, but dropping an issue here just in case anyone else runs into this because it's incredibly difficult to solve.

If you see this issue while following the SwiftUI installation for a full screen ad:
"The provided view controller is not being presented"
it may be due to declaring 
`@Environment(\.openURL) var openURL`

That's it.  That's the bug.  If you remove that line and replace its usage with something like 
`UIApplication.shared.open(Constants.appStoreURL)`
the full screen ad shows up. I have no idea why this happens, but this is the last place you would expect to find the problem.

**Comment by NVentimiglia** [comment]
@abear247 did you need to modify out samples to get it to work right? If so, let me know which ones and I can get the change done.

**Comment by abear247** [comment]
@NVentimiglia This appears to be a direct problem with google ads itself.  We estimate that we are not seeing ads ~50% of the time because of this bug.  It's pretty easy to recreate using the `openURL` but it's not the only way to trigger it.

If I just add this view
```

struct BrokenView: View {
    @State private var showAnotherView = false
    // comment this next line out and the ad will load, keep it in and it will not.  This behaviour of being unable to find the
    @Environment(\.openURL) var openURL
    
    var body: some View {
        NavigationView {
            Button {
                showAnotherView = true
            } label: {
                Text("tap on this")
            }
        }.sheet(isPresented: $showAnotherView, content: {
            InterstitialContentView(navigationTitle:  "broken")
        })
    }
}
```

and use it in the extension
```
extension MenuItem {
  @ViewBuilder
  private func viewForType() -> some View {
    switch self {
    case .banner:
      BannerContentView(navigationTitle: self.rawValue)
    case .interstitial:
        BrokenView()
    case .native:
      NativeContentView(navigationTitle: self.rawValue)
    case .rewarded:
      RewardedContentView(navigationTitle: self.rawValue)
    case .rewardedInterstitial:
      RewardedInterstitialContentView(navigationTitle: self.rawValue)
    }
  }
}
```

The ad will never show on the simulator, on devices its a 50/50 chance.

**Comment by NVentimiglia** [comment]
@abear247 

I am still confused at to what the issue is. None of our samples include the line `@Environment(\.openURL) var openURL` as far as I can see.

**Comment by abear247** [comment]
Yeah they don't, they only cover the most basic scenarios.  The google ads framework starts to break once used beyond these simple contexts.
